tests/gui/test_widgets.py

```diff
@@ -950,6 +950,70 @@ def test_MainView_set_conversation(mocker):
     mv.view_layout.addWidget.assert_called_once_with(mock_widget)
 
 
+def test_MainView__delete_source(mocker):
+    """
+    Ensure that the source and conversation provided by source uuid are deleted.
+    """
+    source = factory.Source(uuid="123")
+    conversation_wrapper = SourceConversationWrapper(source, mocker.MagicMock())
+    mv = MainView(None)
+    mv.source_conversations = {}
+    mv.source_conversations["123"] = conversation_wrapper
+    mv.source_list.controller = mocker.MagicMock()
+    mv.source_list.update([source])
+    assert not mv.source_list.source_items == {}
+
+    mv._delete_source("123")
+
+    assert mv.source_conversations == {}
+    assert mv.source_list.source_items == {}
+
+
+def test_MainView__delete_source_with_multiple_sources(mocker):
+    """
+    Ensure that the source and conversation provided by source uuid are deleted.
+    """
+    source1 = factory.Source(uuid="abc")
+    source2 = factory.Source(uuid="123")
+    mv = MainView(None)
+    mv.source_conversations = {}
+    mv.source_conversations["123"] = SourceConversationWrapper(source1, mocker.MagicMock())
+    mv.source_conversations["123"] = SourceConversationWrapper(source2, mocker.MagicMock())
+    mv.source_list.controller = mocker.MagicMock()
+    mv.source_list.update([source1, source2])
+    assert not mv.source_list.source_items == {}
+
+    mv._delete_source("123")
+    mv._delete_source("abc")
+
+    assert mv.source_conversations == {}
+    assert mv.source_list.source_items == {}
```

Comment:
I like how these tests use mocks to simplify the setup but test the outcome of the deletions! (i.e. don't mock the object under test) :star: